libs/tracker/shared/src/lib/read-config.ts

```
@@ -2,16 +2,21 @@ import { existsSync, readFileSync } from 'fs';
 import { join } from 'path';
 import { Config } from './models/config';
 
-export function readConfig(
-  rootDirectory: string,
-  optionsPath?: string
-): Config {
-  const path =
-    optionsPath && existsSync(join(rootDirectory, optionsPath))
-      ? join(rootDirectory, optionsPath)
-      : join(rootDirectory, 'hd-tracker', 'config.json');
+export function readConfig(rootDirectory: string, optionsPath?: string): Config {
+  let path = join(rootDirectory, 'hd-tracker', 'config.json');
+  if (optionsPath) {
+    if (existsSync(join(rootDirectory, optionsPath))) {
+      path = join(rootDirectory, optionsPath);
+    } else {
+      console.log('Provided config file not found at path', join(rootDirectory, optionsPath));
+    }
+  }
 
-  const contents = readFileSync(path).toString('utf-8');
-
-  return JSON.parse(contents);
+  try {
+    const contents = readFileSync(path, 'utf-8');
+    return JSON.parse(contents);
+  } catch (error) {
+    console.error('Error reading config file:', path);
+    process.exit(1);
```

Nit: Are we ok exiting here or should we throw an error and let the caller handle it and then exit?